File: BlurEffectDemo-OC/BlurEffectDemo-OC/BlurEffectViewController.m
//
//  BlurEffectViewController.m
//  BlurEffectDemo-OC
//
//  Created by Fourfire on 2020/9/4.
//  Copyright © 2020 DurianFourfire. All rights reserved.
//

#import "BlurEffectViewController.h"

@interface BlurEffectViewController ()

@property (nonatomic, strong) UIVisualEffectView *effectView;
@property (nonatomic, strong) UISegmentedControl *segmentedControl;


@end

@implementation BlurEffectViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor whiteColor];
    [self setEdgesForExtendedLayout:UIRectEdgeNone];
    self.navigationItem.title = @"UIVisualEffectView + UIBlurEffect";
    
    UIImageView *imageView = [[UIImageView alloc] initWithFrame:self.view.frame];
    imageView.image = [UIImage imageNamed:@"Bristle Grass.jpg"];
    imageView.contentMode = UIViewContentModeScaleAspectFill;
    imageView.clipsToBounds = true;
    imageView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    [self.view addSubview:imageView];
    
    UIVisualEffectView *effectView = [[UIVisualEffectView alloc] initWithFrame:imageView.frame];
    effectView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    [self.view addSubview:effectView];
    _effectView = effectView;
    
    UIBlurEffect *blurEffect = [UIBlurEffect effectWithStyle:UIBlurEffectStyleDark];
    _effectView.effect = blurEffect;
    
    UISegmentedControl *segmentedControl = [[UISegmentedControl alloc] initWithItems:@[@"None",@"ExtraLight",@"Light",@"Dark"]];
    [segmentedControl addTarget:self action:@selector(segmentedControlValueChanged:) forControlEvents:UIControlEventValueChanged];
    [self.view addSubview:segmentedControl];
    segmentedControl.selectedSegmentIndex = 3;
    _segmentedControl = segmentedControl;
}


- (void)viewWillLayoutSubviews
{
    _segmentedControl.frame = CGRectMake(10, CGRectGetMaxY(self.view.frame) - 50 - self.view.safeAreaInsets.bottom, CGRectGetMaxX(self.view.frame) - 20, 30);
}


- (void)segmentedControlValueChanged:(UISegmentedControl *)segmented
{
    UIBlurEffectStyle effectStyle = segmented.selectedSegmentIndex - 1;
    if (effectStyle >= 0) {
        _effectView.hidden = false;
        UIBlurEffect *blurEffect = [UIBlurEffect effectWithStyle:effectStyle];
        _effectView.effect = blurEffect;
    }else{
        _effectView.hidden = true;
    }
}

@end

File: BlurEffectDemo-Swift/BlurEffectDemo-Swift/BlurEffectViewController.swift
//
//  BlurEffectViewController.swift
//  BlurEffectDemo-Swift
//
//  Created by Fourfire on 2020/9/8.
//  Copyright © 2020 zy. All rights reserved.
//

import UIKit

class BlurEffectViewController: UIViewController {
    private var effectView : UIVisualEffectView!
    private var segmentControl : UISegmentedControl!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = .white
        edgesForExtendedLayout = UIRectEdge.init(rawValue: 0)
        navigationController?.navigationBar.isTranslucent = false
        navigationItem.title = "UIVisualEffectView + UIBlurEffect"
        
        let imageView = UIImageView.init(frame: view.frame)
        imageView.image = UIImage.init(named: "Bristle Grass.jpg")
        imageView.contentMode = .scaleAspectFill
        imageView.clipsToBounds = true
        imageView.autoresizingMask = [.flexibleHeight,.flexibleWidth]
        view.addSubview(imageView)
        
        effectView = UIVisualEffectView.init(frame: view.frame)
        effectView.autoresizingMask = [.flexibleHeight,.flexibleWidth]
        view.addSubview(effectView)
        
        segmentControl = UISegmentedControl.init(items: ["None","ExtraLight","Light","Dark"])
        segmentControl.addTarget(self, action: #selector(segmentedControlValueChanged(control:)), for: .valueChanged)
        segmentControl.selectedSegmentIndex = 3
        view.addSubview(segmentControl)
        
        self.segmentedControlValueChanged(control: segmentControl)
    }

    override func viewDidLayoutSubviews() {
        segmentControl.frame = CGRect.init(x: 10, y: view.frame.height - 50 - view.safeAreaInsets.bottom, width: view.frame.width - 20, height: 30)
    }
    
    @objc func segmentedControlValueChanged(control:UISegmentedControl) -> Void{
        let effectStyle:Int = control.selectedSegmentIndex - 1
        if effectStyle >= 0 {
            effectView.isHidden = false
            effectView.effect = UIBlurEffect.init(style: UIBlurEffect.Style(rawValue: effectStyle)!)
        }else{
            effectView.isHidden = true
        }
    }

}

